server/server.js

```diff
@@ -114,36 +113,29 @@ module.exports = function(port, db, githubAuthoriser) {
         var recipientID = conversationInfo.recipient;
         var senderID = req.session.user;
         // Find both the sender and the recipient in the db
-        users.findOne({
+        users.find({
             _id: senderID
-        }, function(err, sender) {
-            if (!err && sender) {
-                users.findOne({
-                    _id: recipientID
-                }, function(err, recipient) {
-                    if (!err && recipient) {
-                        var conversationID = getConversationID(senderID, recipientID);
-                        var participants = [senderID, recipientID].sort();
-                        conversations.insertOne({
-                            _id: conversationID,
-                            participants: participants
-                        }, function(err, result) {
-                            if (!err) {
-                                res.json({
-                                    id: conversationID,
-                                    participants: participants
-                                });
-                            } else {
-                                res.sendStatus(500);
-                            }
-                        });
-                    } else {
-                        res.sendStatus(500);
-                    }
-                });
-            } else {
-                res.sendStatus(500);
+        }).limit(1).next().then(function(sender) {
+            if (!sender) {
+                throw false;
```

Comment:
I'm not sure about this, so it would be good for you to look this up. Should you be throwing here, or should you rejecting the promise? I expected a reject, but the behaviour of them is a little different, so make sure this is correct.